drivers: spi: rv32m1_lpspi: Fix null tx

karstenkoenig opened the pull request:
Initialize the dummy data transfer so spi transfer is defined even for
an undefined tx data buffer.
This is part of a fix for #20589, together with PR #26031

karstenkoenig commented:
Rebased on master and checked again, still works :-)

MaureenHelm reviewed (approved):
thanks @karstenkoenig

karstenkoenig commented:
@henrikbrixandersen I added you as you did the i2c driver for the rv32m1 so are familiar with the rv32m1, hope this is ok :-)

henrikbrixandersen commented on tests/drivers/spi/spi_loopback/boards/rv32m1_vega_ri5cy.conf:
```diff
@@ -6,4 +6,3 @@
 
 CONFIG_SPI_LOOPBACK_DRV_NAME="SPI_0"
 CONFIG_SPI_LOOPBACK_SLAVE_NUMBER=2
-CONFIG_DEBUG_OPTIMIZATIONS=y
```
Should this change be part of this PR? If yes, could you please separate it into its own commit with a descriptive commit message?

karstenkoenig replied:
Done :-)

karstenkoenig committed:
drivers: spi: rv32m1_lpspi: Fix null tx

Initialize the dummy data transfer so spi transfer is defined even for
an undefined tx data buffer. This aligns the rv32m1 spi driver with the
mcux spi driver.

Signed-off-by: Karsten Koenig <karsten.koenig.030@gmail.com>

MaureenHelm commented:
Please rebase to the latest master. I think you're missing the commit that added the buildkite pipeline, hence the check is failing.

karstenkoenig commented:
> Please rebase to the latest master. I think you're missing the commit that added the buildkite pipeline, hence the check is failing.

Done, thanks!
